bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/storage/ldb/KeyValueStorageRocksDB.java

```diff
@@ -548,17 +560,31 @@ public void remove(byte[] key) throws IOException {
         @Override
         public void clear() {
             writeBatch.clear();
+            batchCount = 0;
         }
 
         @Override
         public void deleteRange(byte[] beginKey, byte[] endKey) throws IOException {
             try {
                 writeBatch.deleteRange(beginKey, endKey);
+                countBatchAndFlushIfNeeded();
```

Comment:
We need to discuss whether deleteRange should also be performed within a batch. A deleteRange operation can potentially delete a large number of keys. If we include multiple deleteRange operations within a single batch, I am concerned that it may introduce issues. The previous core dump issue could be related to this.

Reply:
That should be another issue. This PR aims to limit the batch size. We can handle that with another PR.